tools/staramr/staramr_search.xml

```diff
@@ -0,0 +1,233 @@
+<tool id="staramr_search" name="staramr" version="0.2.0">
+    <description>Scans genome assemblies against the ResFinder and PointFinder databases searching for AMR genes.</description>
+    <requirements>
+        <requirement type="package" version="0.2.0">staramr</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        #set $named_genomes = ''
+        #for $genome in $genomes
+            #set $_named_genome = '"{}.fasta"'.format($genome.element_identifier)
+            ln -s "$genome" $_named_genome &&
+            #set $named_genomes = $named_genomes + ' ' + $_named_genome
+        #end for
+
+        #set $excel_proper_extension = '"{}.xlsx"'.format($excel)
+        ln -s "$excel" $excel_proper_extension &&
+
+        staramr search
+
+        --nprocs "\${GALAXY_SLOTS:-1}"
+
+        --pid-threshold $advanced.pid_threshold
+        --percent-length-overlap-resfinder $advanced.plength_resfinder
+        --percent-length-overlap-pointfinder $advanced.plength_pointfinder
+
+        $advanced.report_all_blast
+        $advanced.exclude_negatives
+        $advanced.exclude_resistance_phenotypes
+
+        --output-summary $summary
+        --output-resfinder $resfinder
+        #if $use_pointfinder.enable
+            --output-pointfinder $pointfinder
+        #end if
+        --output-settings $settings
+        --output-excel $excel_proper_extension
+
+        --output-hits-dir staramr_hits
+
+        #if $use_pointfinder.enable
+            --pointfinder-organism $use_pointfinder.organism
+        #end if
+
+        $named_genomes
+    ]]></command>
+    <inputs>
+        <param type="data" name="genomes" format="fasta" multiple="true"/>
```

Comment:
The `multiple="true"` is used to do reductions on a collection of files in a single execution of a tool (e.g., go from **many inputs => one output** with a single instance of `staramr`) - http://planemo.readthedocs.io/en/latest/writing_advanced.html#processing-lists-reductions.  I do this with `staramr` since, even though it processes many files, it already has a built-in method for using many threads and it's pretty quick so it makes sense to do the parallel processing internally to `staramr`.

If, instead, you want the output to be **many inputs => many outputs**, each input file being processed by a separate instance of the tool, then you wouldn't include `multiple="true"`.  This is done with tools such as MentaLiST since there is no built-in parallel processing method and it takes a while to run MentaLiST.  So, excluding `multiple=true` lets us do parallel processing across our cluster, but then we have to run a separate tool to do the reduction afterwards (for MentaLiST, we run a tool to merge all the MLST results into a single table).

I hope this makes sense.

Reply:
For when you would use `multiple=true` vs. explicitly defining a collection, I think they're both valid ways.  I don't know how well `multiple=true` works with paired-end data though (not an issue with FASTA files).  It's probably the preferred method since it's simpler, unless if many other input options need to be used.

Reply:
Thanks for the in-depth explanation! That answers my question of when to use one method over the other.